### PR TITLE
resource_metering: fix the enabled reporter doesn't initialize client during starting server

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs <crazycs520@gmail.com>

### What problem does this PR solve?

As the title said. If the config `resource-metering` is enabled when the server started, the `resource-metering` won't  work since it's client doesn't init.

### Check List 

Tests 

- Manual test

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by crazycs520** [comment]
@zhongzc PTAL

File: components/resource_metering/src/reporter.rs [hunk 1]
```diff
@@ -110,13 +114,13 @@ impl Runnable for ResourceMeteringReporter {
                 if !new_config.should_report() {
                     self.client.take();
                     self.cpu_records_collector.take();
+                    self.config = new_config;
```
Comment (zhongzc) [inline components/resource_metering/src/reporter.rs:117]: why moving the outer to here?

Comment (crazycs520) [reply components/resource_metering/src/reporter.rs:117]: since the `if` and `else` branch both need this, but the order is different, so I move to here.

**Comment by zhongzc** [comment]
I have an idea. How about lazily initializing the client at checking `should_report` so we don't need to do that during config change?

**Review by zhongzc — APPROVED** [review]

**Review by zhongzc — CHANGES_REQUESTED** [review]
test failed

**Review by zhongzc — APPROVED** [review]

**Review by breeswish — APPROVED** [review]

**Comment by crazycs520** [comment]
should cherry-pick? @zhongzc @breeswish

**Comment by crazycs520** [comment]
> cherry pick to release-5.2 failed

release-5.2 already has this fix.

**Pull request merged**